### PR TITLE
Auth system

### PR DESCRIPTION
**Pull request by DirtNap**
This is a working auth and user mgmt framework.

### CODE REVIEW

File: letterbucket/__init__.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# Create or import views needed by the app
+default_view = flask.Blueprint('letterbucket', __name__,
+                               template_folder='templates',  static_folder='static',
+                               static_url_path='', url_prefix='')
+from letterbucket.account import view as account_view
+from letterbucket.content import view as content_view
+
+
+def create_application(config_file=None, config_object=None):
+    """Factory for creating applications
```
Comment (DirtNap) [inline letterbucket/__init__.py]: Comment should be more detailed

File: letterbucket/templates/base_page.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  {% endblock title %}
+  <meta name="description" content="LetterBucket - A Liteweight Content Manager" />
+  <meta name="author" content="Michael Donnelly, Nicole Freeman, Bryan Cole, Michel Erb, Ryan Goldstone" />
+  {% block script %}<!-- Extend this block to completely replace script elements -->
+  {% block page_script %}<!-- Extend this block to add to the scripts -->
+  {% endblock page_script %}
+  {% endblock script %}
+</head>
+<body>
+  <!-- {{ current_user }} -->
```
Comment (xorinox) [inline letterbucket/templates/base_page.html]: Please remove debug comment

Comment (DirtNap) [inline letterbucket/templates/base_page.html]: Resolved in 810da583730228b4e48bd936d5b38cc1f00f1f48

File: letterbucket/static/style/letterbucket.css [hunk 3]
```diff
@@ -0,0 +1,2 @@
+li.flashedMessageError { color: red; font-weight: 800; }
+li.flashedMessageMessage { font-weight: 800; }
```
Comment (xorinox) [inline letterbucket/static/style/letterbucket.css]: Add a newline

File: letterbucket/account/users_test.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                                  test_user_email,
+                                  test_user_name,
+                                  test_user_password)
+            new_user.Persist()
+            self.assertEqual([new_user], users.User.query.all(), 'User should be in the db.')
+            second_user = users.User(u'foo',
+                                     test_user_email,
+                                     test_user_name,
+                                     test_user_password)
+            self.assertRaises(IntegrityError, second_user.Persist)
```
Comment (xorinox) [inline letterbucket/account/users_test.py]: Add test message

File: letterbucket/account/__init__.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+                              flask.request.form['email'],
+                              flask.request.form['name'],
+                              flask.request.form['password'])
+        new_user.Persist()
+        # Now create and persist the blog.
+        new_blog = blogs.Blog(flask.request.form['blog_path'],
+                              flask.request.form['blog_name'],
+                              new_user)
+        new_blog.Persist()
+        flask.flash('Account Created.  Please Log In')
```
Comment (xorinox) [inline letterbucket/account/__init__.py]: Remove one space

File: letterbucket/static/style/letterbucket.css (hunk 3 above)
Comment (DirtNap) [inline letterbucket/static/style/letterbucket.css]: Resolved in 131f37279a67cbc2a06174c037e397739f0a8e17

File: letterbucket/account/users_test.py (hunk 4 above)
Comment (DirtNap) [inline letterbucket/account/users_test.py]: This is resolved in 3af19a4d257251b5b237cb143161fac9542fc804 -- however, it turns out that `assertRaises` does not support the message argument.  I took the constraint error pattern and moved it into the `_AssertConstraintError` method.  This has the added advantage of avoiding repeating the db rollback, and targeting particular exceptions and columns.

However, in the `testUserPassword` method I needed to use `assertRaises` again, but it wasn't an interesting enough case to create a method for, so those do not support messages still.  In this case I added a comment above each.

File: letterbucket/account/__init__.py (hunk 5 above)
Comment (DirtNap) [inline letterbucket/account/__init__.py]: Done in 4715554331337ee0ffb5ac2352e3d93dc3fb1b98

**Pull request merged**